efficientdet/third_party/tools/nvgpu.py

```diff
@@ -96,3 +96,39 @@ def gpu_info():
     return XmlDictConfig(root)
   except FileNotFoundError:
     return None
+
+
+def gpu_memory_util_message():
+  """Provide information about GPUs."""
+  gpu_info_d = gpu_info()
+  if gpu_info_d is not None:
+    mem_used = gpu_info_d['gpu']['fb_memory_usage']['used']
+    mem_total = gpu_info_d['gpu']['fb_memory_usage']['total']
+    mem_util = commonsize(mem_used) / commonsize(mem_total)
+    logstring = ("GPU memory used: {} = {:.1%} ".format(mem_used, mem_util) +
+                 "of total GPU memory: {}".format(mem_total))
+    return logstring
+  return None
+
+
+def commonsize(inp):
+  """Convert all to MiB."""
```

Comment:
how about a more informative name such as 'input_size'?  Similar, you can rename 'inp_' to 'converted_size' or 'output_size'

Reply:
Thanks @mingxingtan 
I added some additional details to the comments regarding why "Add" and what other options could be used.
I understand that the answer at this moment is not fully satisfactory. My thinking was that first we add the functionality of gradient checkpointing to the repo and then the community would help to optimize its parameters. ["Add"] was an educated guess that was made without full analysis of the network. Out of luck, it worked reasonably well. I am looking at some graph optimization algorithms to see if the choice of nodes to checkpoint can be improved.